### PR TITLE
ML course notebooks 5-7.

### PR DESCRIPTION
**Pull request by dansbecker**
Broadly speaking, this is trivial changes to earlier notebooks, but substantive changes to steps 5-7 of the ML course.

- Added ex5 and ex6 py files. Step 7 has no tests to be run. Added ipynb files and testing for 5-7.
- Moved input csv file to have same relative path in these notebooks as they'll have in Kernels. Changed the paths in all notebooks.
- The 7th tutorial is about submitting to a comp.  This still isn't explained as completely as should be, but the details are likely to change quickly with changes in the kernels editor and viewer.

### CODE REVIEW

File: learntools/machine_learning/nbs/ch6.ipynb [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+     "text": [
+      "Validation MAE for Random Forest Model: \n"
+     ]
+    }
+   ],
+   "source": [
+    "from sklearn.ensemble import RandomForestRegressor\n",
+    "\n",
+    "# Define the model. Set random_state to 1\n",
+    "rf_model = _\n",
```
Comment (colinmorris) [inline learntools/machine_learning/nbs/ch6.ipynb]: I'm still somewhat wary of giving them starter code that can't be run all the way through without exceptions, though it's not something I feel super-strongly about.

File: learntools/machine_learning/nbs/ch6.ipynb [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    "\n",
+    "# fit your model\n",
+    "_\n",
+    "\n",
+    "# Calculate the mean absolute error of your Random Forest model on the validation data\n",
+    "rf_val_mae = _\n",
+    "\n",
+    "print(\"Validation MAE for Random Forest Model: {}\".format(rf_val_mae))\n",
+    "\n",
+    "# step_1.check()"
```
Comment (colinmorris) [inline learntools/machine_learning/nbs/ch6.ipynb]: Also, I've been following the convention of always starting with the call to `check()` uncommented.

File: learntools/machine_learning/nbs/ch7.ipynb [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    "\n",
+    "print(\"Validation MAE for Random Forest Model: {:,.0f}\".format(rf_val_mae))\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Submitting Your Model to a Competition\n",
+    "Data science isn't always this easy. But replacing the decision tree with a Random Forest is going to be an easy win."
```
Comment (colinmorris) [inline learntools/machine_learning/nbs/ch7.ipynb]: Leftover copy-paste from previous exercise notebook?

Also, the code cell below where we print columns seems to come out of nowhere. Is that supposed to be there?

**Comment by colinmorris** [comment]
Looks good to me. I didn't actually attempt to run the exercise code this time (I think it might make sense for me to try running through the whole track once you think it's ready?)

File: learntools/machine_learning/nbs/ch6.ipynb (hunk 1 above)
Comment (dansbecker) [reply learntools/machine_learning/nbs/ch6.ipynb]: Unless I've screwed something up when testing, this runs all the way through without exceptions.

`rf_model = _` is the same as
`rf_model = ''"`

I wasn't aware of this until recently, though I'm guessing it is related to our ability to do
`("whatever" for _ in range(10))`

Comment (dansbecker) [reply learntools/machine_learning/nbs/ch6.ipynb]: Oh. I see now that it fails the assertion that rf_val_mae is a number when it hits q1.check().  I think that's less bad then the type of exceptions resulting in ugly and hard-to-understand stack traces.

**Comment by dansbecker** [comment]
Having you run everything from beginning to end sounds perfect.  Let's talk about doing that on Kaggle before this goes live to Learn. I think that will be relatively easy and worthwhile.

**Pull request merged**